### PR TITLE
[KB-195] Methodology section

### PR DESCRIPTION
**Pull request by Pattrigue**

### CODE REVIEW

**Review by chhoumann — COMMENTED** [review]
I think the introduction is just too vague right now. Could you make an outline for what the entire experiments / methodology section should say? Bullet points are fine, but it's nigh impossible to give feedback when it isn't clear to me why some things are left vague, some things are left out, and why it's being structured the way it is.

I understand that we wanted to postpone writing the hook until our problem definition is finalized, so let's ignore that for now.
`State of the art machine learning methodologies` is too vague - we've been criticized for this before.
`We tested various methodologies` is too vague as well. I understand that this is the introduction, but it could be more specific. And it's left out in the `In this section,` as well as the subsequent subsections.

I'm also unsure why the definitions of the models appear in this section. I agree with your comment `maybe put these in the background section instead for the five models, however that might come out of the blue??`.
I think it's preferable to focus on getting this section down before moving on to writing that? We can always describe the algorithms, they don't change. But we should probably get our experiments down in writing before the details start getting blurry.
Meaning, I don't think describing the algorithms is part of this task.

File: report_thesis/src/sections/methodology.tex [hunk 1]
```diff
@@ -1,23 +1,54 @@
...
+For this purpose, we considered the following preprocessing techniques: Min-Max Scaling, Standard Scaling, Robust Scaling, MaxAbs Scaling, Quantile Transformer and Power Transformer. 
+Additionally, we also considered dimensionality reduction techniques such as PCA and KernelPCA.
+Through experimentation, we identified which preprocessing techniques were most effective for each model and oxide. 
+Realizing the potential of this, we decided to further investigate how the predictive accuracy and robustness could be improved by combining the strengths of these models and preprocessing techniques.
+From our investigation, we determined that the stacking ensemble method seemed very promising and novel in the LIBS analysis field, leading us to select it as the primary method for our study.
+
+Stacking ensemble is analogous to the methodologies employed in the original MOC pipeline, which also tailored predictions for each oxide by blending outputs from the PLS-SM and ICA phases, variably weighting the influence of the ICA predictions depending on the oxide.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/methodology.tex:44]: Needs cite

File: report_thesis/src/sections/methodology.tex [hunk 2]
```diff
@@ -1,23 +1,54 @@
...
-To assess the robustness of our models, we will consider the standard deviation of prediction errors across each oxide and test instance. This metric measures the variability of the prediction errors and provides insight into the consistency of the model's performance. It is defined as:
-
-\begin{equation}
-    \sigma_{error} = \sqrt{\frac{1}{n-1} \sum_{i=1}^{n} (e_i - \bar{e})^2},
-\end{equation}
-
-where $e_i = y_i - \hat{y}_i$ and $\bar{e}$ is the mean error. A lower standard deviation indicates better robustness.
-
-By using these two metrics, we aim to evaluate model performance in terms of both accuracy and robustness, which are crucial for the reliable prediction of major oxide compositions from \gls{libs} data.
+In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed in the literature, outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:2]:
```suggestion
In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data. 
```

File: report_thesis/src/sections/methodology.tex [hunk 3]
```diff
@@ -1,23 +1,54 @@
...
-
-By using these two metrics, we aim to evaluate model performance in terms of both accuracy and robustness, which are crucial for the reliable prediction of major oxide compositions from \gls{libs} data.
+In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed in the literature, outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data. 
+Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
+We first present a description of the datasets used, how it was prepared and how it was split for model training. Then the preprocessing and model selection process is described followed by the experimental setup and the evaluation metrics employed. Finally, we describe our validation testing procedures and how the final model evaluations were conducted to ensure unbiased results.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:4]:
```suggestion
We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
```

File: report_thesis/src/sections/methodology.tex [hunk 4]
```diff
@@ -1,23 +1,54 @@
...
+
+\subsection{Data Prepraration}
+Similarly to our previous work \cite{p9_paper}, we used the publicly available \gls{ccs} data. 
+CCS refers to LIBS data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:8]:
```suggestion
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 
```

File: report_thesis/src/sections/methodology.tex [hunk 5]
```diff
@@ -1,23 +1,54 @@
...
+
+\subsection{Data Prepraration}
+Similarly to our previous work \cite{p9_paper}, we used the publicly available \gls{ccs} data. 
+CCS refers to LIBS data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 
+The full description of this preprocessing can be found in \citet{wiensPreFlight3}.
+
+While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing to be useful. Specifically, the data will still have noise at the boundaries of the spectrometer ranges, which have to be masked out. The data will also contain negative values, which are not physically meaningful and have to be set to zero.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:11]:
```suggestion
While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing to be useful. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.
```

File: report_thesis/src/sections/methodology.tex [hunk 6]
```diff
@@ -1,23 +1,54 @@
...
+For a more detailed description of the data and the preprocessing steps, we refer the reader to \citet{p9_paper}, since the data preparation follows the same procedures detailed in that work.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:12]:
```suggestion
For a more detailed description of the data and the preprocessing steps, we refer the reader to \citet{p9_paper}, as data preparation follows the same procedure detailed in that work.
```

File: report_thesis/src/sections/methodology.tex [hunk 7]
```diff
@@ -1,23 +1,54 @@
...
+
+Once preprocessed, the data was conventionally split into training and testing sets. 
+However, as detailed in \citet{p9_paper} and \cite{cleggRecalibrationMarsScience2017}, care has to be taken in order to ensure that the training set is constructed such that it has sufficient samples containing extreme values.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:15]:
```suggestion
As detailed in \citet{p9_paper} and \citet{cleggRecalibrationMarsScience2017}, it is essential to include samples with extreme element compositions in the training set to ensure the model can generalize effectively.
```

File: report_thesis/src/sections/methodology.tex (hunk 1 above)
Comment (Pattrigue) [reply report_thesis/src/sections/methodology.tex:44]: ok 👍

File: report_thesis/src/sections/methodology.tex [hunk 8]
```diff
@@ -1,23 +1,54 @@
...
+To incorporate these extreme values, the training set was constructed by including the two highest and two lowest extreme samples from each oxide category.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:16]:
```suggestion
To incorporate these extreme values, the training set was constructed to include the two highest and two lowest samples from each oxide category.
```

File: report_thesis/src/sections/methodology.tex [hunk 9]
```diff
@@ -1,23 +1,54 @@
...
+Following the inclusion of extreme samples, we adjusted the size of the test split to accommodate the reduced pool of normal samples.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:17]: Think this needs to be phrased, unsure what exactly you mean by this.

File: report_thesis/src/sections/methodology.tex [hunk 10]
```diff
@@ -1,23 +1,54 @@
...
+Following the inclusion of extreme samples, we adjusted the size of the test split to accommodate the reduced pool of normal samples. 
+This adjustment was calculated by determining the number of extreme samples, subtracting this from the total sample count, and recalculating the test size proportionally. 
+This ensures that the proportion of the test set remains appropriate relative to the now smaller dataset. 
+Subsequently, we conducted the train-test split, where the normal samples --- those not categorized as extreme --- were divided according to the newly adjusted test size, with the extreme samples included to ensure exposure to varied data points.
+
+\subsection{Model and Preprocessing Selection}
+For the initial investigative experiments we selected a range of models for further exploration, namely Support Vector Regression (SVR), Gradient Boosting Regression (GBR), Partial Least Squares Regression (PLS), eXtreme Gradient Boosting (XGBoost), Natural Gradient Boosting (NGBoost), Extra Trees Regression (ETR), Elastic Net (ENet), and Stochastic Gradient Descent (SGD).
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:23]:
```suggestion
For the initial investigative experiments, we selected a range of models for further exploration, namely \gls{svr}, \gls{gbr}, \gls{pls}, \gls{xgboost}, \gls{ngboost}, \gls{etr}, \gls{enet}, and \gls{sgd}.
```

File: report_thesis/src/sections/methodology.tex [hunk 11]
```diff
@@ -1,23 +1,54 @@
...
+This adjustment was calculated by determining the number of extreme samples, subtracting this from the total sample count, and recalculating the test size proportionally. 
+This ensures that the proportion of the test set remains appropriate relative to the now smaller dataset. 
+Subsequently, we conducted the train-test split, where the normal samples --- those not categorized as extreme --- were divided according to the newly adjusted test size, with the extreme samples included to ensure exposure to varied data points.
+
+\subsection{Model and Preprocessing Selection}
+For the initial investigative experiments we selected a range of models for further exploration, namely Support Vector Regression (SVR), Gradient Boosting Regression (GBR), Partial Least Squares Regression (PLS), eXtreme Gradient Boosting (XGBoost), Natural Gradient Boosting (NGBoost), Extra Trees Regression (ETR), Elastic Net (ENet), and Stochastic Gradient Descent (SGD). 
+We also included regular neural network (NN) and convolutional neural networks (CNN) in this phase of experimentation.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:24]:
```suggestion
We also included regular \gls{ann} and \gls{cnn} in this phase of experimentation. 
```

File: report_thesis/src/sections/methodology.tex [hunk 12]
```diff
@@ -1,23 +1,54 @@
...
+This ensures that the proportion of the test set remains appropriate relative to the now smaller dataset. 
+Subsequently, we conducted the train-test split, where the normal samples --- those not categorized as extreme --- were divided according to the newly adjusted test size, with the extreme samples included to ensure exposure to varied data points.
+
+\subsection{Model and Preprocessing Selection}
+For the initial investigative experiments we selected a range of models for further exploration, namely Support Vector Regression (SVR), Gradient Boosting Regression (GBR), Partial Least Squares Regression (PLS), eXtreme Gradient Boosting (XGBoost), Natural Gradient Boosting (NGBoost), Extra Trees Regression (ETR), Elastic Net (ENet), and Stochastic Gradient Descent (SGD). 
+We also included regular neural network (NN) and convolutional neural networks (CNN) in this phase of experimentation. 
+This selection was guided primarily by the literature review but also by exploratory intuition,
+aiming to discover potentially innovative applications and performances within our specific dataset.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:26]:
```suggestion
This selection was guided primarily by the literature review but also by exploratory intuition, aiming to discover potentially innovative applications and performances within our specific dataset.
```

File: report_thesis/src/sections/methodology.tex [hunk 13]
```diff
@@ -1,23 +1,54 @@
...
+
+Our literature review highlighted various approaches and their effectiveness in handling the challenges associated with predicting major oxide compositions from LIBS data.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:28]:
```suggestion
Our literature review highlighted various approaches and their effectiveness in handling the challenges associated with predicting major oxide compositions from \gls{libs} data. 
```

File: report_thesis/src/sections/methodology.tex [hunk 14]
```diff
@@ -1,23 +1,54 @@
...
+
+All models demonstrated robust performance; however, \gls{gbr}, \gls{svr}, \gls{xgboost}, \gls{etr}, and \gls{pls} consistently excelled across all oxides. 
+Notably, we observed model-specific strengths for certain oxides—SVR was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}. 
+This differential performance prompted us to explore architectural frameworks that could systematically capitalize on the strengths of each model for specific oxides.
+
+A particularly exciting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts. 
+Inspired by these ideas, we explored the possibility of improving model performance by optimizing the preprocessing chain for each model, per oxide, in order to determine which normalization methods would be most beneficial.
+For this purpose, we considered the following preprocessing techniques: Min-Max Scaling, Standard Scaling, Robust Scaling, MaxAbs Scaling, Quantile Transformer and Power Transformer. 
+Additionally, we also considered dimensionality reduction techniques such as PCA and KernelPCA.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:39]:
```suggestion
Additionally, we also considered dimensionality reduction techniques such as \gls{pca} and \gls{kernel-pca}.
```

File: report_thesis/src/sections/methodology.tex (hunk 1 above)
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:44]:
```suggestion
Stacking ensemble is analogous to the methodologies employed in the original \gls{moc} pipeline, which also tailored predictions for each oxide by blending outputs from the \gls{pls1-sm} and \gls{ica} phases, variably weighting the influence of the \gls{ica} predictions depending on the oxide.
```

File: report_thesis/src/sections/methodology.tex [hunk 15]
```diff
@@ -1,23 +1,54 @@
...
+We also included regular neural network (NN) and convolutional neural networks (CNN) in this phase of experimentation. 
+This selection was guided primarily by the literature review but also by exploratory intuition,
+aiming to discover potentially innovative applications and performances within our specific dataset.
+
+Our literature review highlighted various approaches and their effectiveness in handling the challenges associated with predicting major oxide compositions from LIBS data. 
+For instance, Anderson et al. discussed the use of multiple regression models, finding that different models excelled with specific oxides, which informed our model-specific approach. 
+Song et al. presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. Rezaei et al. demonstrated the beneficial impact of dimensionality reduction techniques like PCA, which we considered essential for managing our high-dimensional LIBS data.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:30]:
```suggestion
\citet{song_DF-K-ELM} presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. \citet{rezaei_dimensionality_reduction} demonstrated the beneficial impact of dimensionality reduction techniques like \gls{pca}, which we considered essential for managing our high-dimensional \gls{libs} data.
```

File: report_thesis/src/sections/methodology.tex [hunk 16]
```diff
@@ -1,23 +1,54 @@
...
+
+All models demonstrated robust performance; however, \gls{gbr}, \gls{svr}, \gls{xgboost}, \gls{etr}, and \gls{pls} consistently excelled across all oxides. 
+Notably, we observed model-specific strengths for certain oxides—SVR was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:33]:
```suggestion
Notably, we observed model-specific strengths for certain oxides --- \gls{svr} was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}. 
```

File: report_thesis/src/sections/methodology.tex [hunk 17]
```diff
@@ -1,23 +1,53 @@
...
+We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
+
+\subsection{Data Prepraration}
+Similarly to our previous work \cite{p9_paper}, we used the publicly available \gls{ccs} data. 
+\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 
+The full description of this preprocessing can be found in \citet{wiensPreFlight3}.
+
+While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.
```
Comment (Pattrigue) [inline report_thesis/src/sections/methodology.tex:11]:
```suggestion
While the \gls{ccs} data is in a more suitable form for quantitative analysis, it still requires further preprocessing. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.
```

File: report_thesis/src/sections/methodology.tex [hunk 18]
```diff
@@ -1,2 +1,134 @@
...
+240.81100 & 6.4026649e+15 & 4.0429349e+15 & $\cdots$  & 1.7922483e+15 & 1.7126615e+15 & 1.9892956e+15 & 1.7561699e+15 \\
+240.86501 & 3.8557462e+12 & 2.2923680e+12 & $\cdots$  & 1.1355429e+12 & 8.6930379e+11 & 7.8172542e+11 & 7.2805052e+11 \\
+$\vdots$  & $\vdots$      & $\vdots$      & $\cdots$  & $\vdots$      & $\vdots$      & $\vdots$      & $\vdots$      \\
+905.38062 & 1.8823427e+08 & 58500403.     & $\cdots$  & -8449286.6    & 8710775.0     & 4.0513312e+09 & 5.2188327e+09 \\
+905.57349 & 1.9864713e+10 & 1.2956832e+10 & $\cdots$  & 1.9785415e+10 & 7.1994239e+09 & 1.1311150e+10 & 1.2201224e+10 \\
+\bottomrule
+\end{tabular}
+\caption{Example of CCS data for a single target location (from \citet{p9_paper})}
+\label{tab:ccs_data_example}
+\end{table*}
```
Comment (chhoumann) [inline report_thesis/src/sections/methodology.tex:27]: Can see we're using the label anywhere - need to reference this table?

File: report_thesis/src/sections/methodology.tex [hunk 19]
```diff
@@ -1,2 +1,134 @@
...
+These negative values are not physically meaningful, since you cannot have negative light intensity.
+Similar to the noisy edges, these negative values are also masked by zeroing out the values.
+
+We transpose the data so that each row represents a target location and each column represents a wavelength feature.
+Each location is now represented as a vector of wavelengths, with the corresponding average intensity values for each wavelength.
+These vectors are then concatenated to form a matrix, where each row represents a target location and each column represents a wavelength feature.
+
+For each target, we have a corresponding set of major oxide compositions in weight percentage (wt\%).
+These compositions are used as the target labels for the machine learning models.
+An excerpt of this data is shown in Table \ref{tab:composition_data_example}.
```
Comment (chhoumann) [inline report_thesis/src/sections/methodology.tex:49]: Would it be easier for the reader to understand this part if we walked through the table? Just take one row and explain each column? Saying "The target, spectrum, and sample names are artifacts of the measurement progress. We primarily regard the sample name. The eight oxides are SiO2, ..., and serve as our ground truth values. The MOC total column is unused in our analysis" or something?

File: report_thesis/src/sections/methodology.tex [hunk 20]
```diff
@@ -1,2 +1,134 @@
...
+	$\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ \\
+\midrule
+\end{tabular}
+\caption{Excerpt from the final dataset (values have been rounded to two decimal places for brevity).}
+\label{tab:final_dataset_example}
+\end{table*}
+
+
+\subsection{Model and Preprocessing Selection}
+For the initial investigative experiments, we selected a range of models for further exploration, namely \gls{svr}, \gls{gbr}, \gls{pls}, \gls{xgboost}, \gls{ngboost}, \gls{etr}, \gls{enet}, and \gls{sgd}.
```
Comment (chhoumann) [inline report_thesis/src/sections/methodology.tex:98]: Need to ref back to related works to explain why we picked these? I can see we're kinda covering it in the next few lines, but maybe be really precise as to why we picked these? Not easy, of course. It's a while ago we actually did that work.

Also, the SuperCam model selection paper isn't mentioned here, which I believe was the inspiration for a good few models.

File: report_thesis/src/sections/methodology.tex [hunk 21]
```diff
@@ -1,2 +1,134 @@
...
+\citet{song_DF-K-ELM} presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. \citet{rezaei_dimensionality_reduction} demonstrated the beneficial impact of dimensionality reduction techniques like \gls{pca}, which we considered essential for managing our high-dimensional \gls{libs} data.
+
+All models demonstrated robust performance; however, \gls{gbr}, \gls{svr}, \gls{xgboost}, \gls{etr}, and \gls{pls} consistently excelled across all oxides.
+Notably, we observed model-specific strengths for certain oxides --- \gls{svr} was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}.
+This differential performance prompted us to explore architectural frameworks that could systematically capitalize on the strengths of each model for specific oxides.
+
+A particularly exciting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts.
```
Comment (chhoumann) [inline report_thesis/src/sections/methodology.tex:110]:
```suggestion
An interesting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts.
```

File: report_thesis/src/sections/methodology.tex [hunk 22]
```diff
@@ -1,2 +1,134 @@
...
+    0        & $\cdots$     & 5.29e+10 & 3.62e+10 & $\cdots$ & 0        & 57.60     & 0.78 & 26.60 & 2.73 & 0.70 & 0.01 & 0.38 & 7.10 & pg7         & pg7\_2013\_11\_07\_161552\_ccs \\
+	$\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ \\
+\midrule
+\end{tabular}
+\caption{Excerpt from the final dataset (values have been rounded to two decimal places for brevity).}
+\label{tab:final_dataset_example}
+\end{table*}
+
+
+\subsection{Model and Preprocessing Selection}
```
Comment (chhoumann) [inline report_thesis/src/sections/methodology.tex:97]: I think, particularly in this subsection, there's a lack of structure and depth. Most of the text _feels_ surface level, and it's kind of all-over-the-place.

It somewhat tracks the story, but skips over a lot of important details regarding the rationale behind our decisions.
And it completely omits the last month and a half of work (optimization framework, for example).

I think we need a lot more depth to make it work, as well as a well-formed structure. We probably need to spell out some outline for what exactly should be said here, as I don't think it's going to be a quick fix.

We did a ton of awesome work this semester, and I'm greedy enough on our behalf to want us to get credit for every single thing we did. Let's not downplay it by being brief.

**Review by chhoumann — COMMENTED** [review]
I think the direction here is great. Most of it is good, especially when considered individually.
That aside, I think there are some lacks in the general structure and depth at this point, which makes it feel like an unfinished section yet.

Let me also encourage you to do a find and replace on `LIBS` → `\gls{libs}`.

File: report_thesis/src/sections/methodology.tex (hunk 22 above)
Comment (Ivikhostrup) [reply report_thesis/src/sections/methodology.tex:97]: Made task for this

File: report_thesis/src/sections/methodology.tex [hunk 23]
```diff
@@ -1,2 +1,134 @@
 \section{Methodology}\label{sec:methodology}
-\textit{We will write an introduction to the methodology section here, as well as add more subsections in the future.}
+In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data.
+Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
```
[the remark this answers is not in the document]
Comment (Ivikhostrup) [reply report_thesis/src/sections/methodology.tex:3]: Made task for this

File: report_thesis/src/sections/methodology.tex [hunk 24]
```diff
@@ -1,2 +1,134 @@
...
+We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
+
+
+\subsection{Data Prepraration}
+Similarly to our previous work \citet{p9_paper}, we used the publicly available \gls{ccs} data.
+\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.
+The full description of this preprocessing can be found in \citet{wiensPreflightCalibrationInitial2013}.
```
[the remark this answers is not in the document]
Comment (Ivikhostrup) [reply report_thesis/src/sections/methodology.tex:10]:
```suggestion
A comprehensive description of this preprocessing procedure is available in \citet{wiensPreflightCalibrationInitial2013}.
```

File: report_thesis/src/sections/methodology.tex [hunk 25]
```diff
@@ -1,2 +1,136 @@
...
+Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
+We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
+
+
+\subsection{Data Preparation}
+Similarly to our previous work \citet{p9_paper}, we used the publicly available \gls{ccs} data from NASA's \gls{pds}~\cite{PDSGeoscienceNode}.
+\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/methodology.tex:9]:
```suggestion
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps, as described by \citet{wiensPreflightCalibrationInitial2013}, such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.
```

**Review by chhoumann — APPROVED** [review]

**Pull request merged**